src/main/java/org/testng/internal/TestNGClassFinder.java

```diff
@@ -297,6 +244,7 @@ private static boolean isNotTestNGClass(Class<?> c, IAnnotationFinder annotation
    * @return true if this class contains TestNG annotations (either on itself
    * or on a superclass).
    */
+  @SuppressWarnings("unchecked")
```

Comment:
Same here

Reply:
Had to add this because `AnnotationHelper.getAllAnnotations()` returns back an array of `Class` whereas the functions within the for loop need `Class<? extends IAnnotation>`.

Reply:
Update the signature of getAllAnnotations if you can.

Reply:
I have already explored that. Doing that basically shifts the warning from one place to the other because we cannot create a generic array for e.g., `new Class<? extends IAnnotation>[] {}`

Reply:
Using a set or a list is possible too 😉

Reply:
@juherr - This is now fixed. Please take a look (Amended the commit and pushed in the changes)